hermes/src/network/p2p.go

```diff
@@ -74,12 +77,34 @@ func RegisterObservationCallback(f C.callback_t, network_id, bootstrap_addrs, li
 	var recoverRerun func()
 
 	routine := func() {
+		// If the current goroutine fails for any reasons, recoverRerun will be
+		// triggered which will sleep for 1 second and rerun the routine.
 		defer recoverRerun()
 
-		// Record the current time
+		// Record the current time for profiling and backoff purposes.
 		startTime = time.Now().UnixNano()
 
-		ctx := context.Background()
+		// Start the datadog profiler. Only starts if an environment is
+		// specified and a datadog API key is provided.
+		if os.Getenv("DD_PROFILING_ENV") != "" && os.Getenv("DD_API_KEY") != "" {
```

Comment:
wouldn't it be better if we pass them by cli arguments (which supports env variable as well)?

Reply:
Yes I'll change it, I added them with changes to the `GO_LOG` which I intended to behave similar to `RUST_LOG` (no arg), this would be better.